server/routes/events.js

```diff
@@ -2,14 +2,251 @@
+
+e.g. if date= is set, and so is month=, then its date OR month because it wouldn't make sense to answer both queries. Here date is more specific, so do date
+
+e.g. month= and year= are set. They work together, so it's month AND year
+
+generally filters of the similar type/category form an OR relationship while filters of different types/topics form AND relationships. aka "when" filters (date/time related) form OR relationships between each other, but "when" filters have an AND relationship with category, company, and featured
+*/
+
+/* alternative syntax for queries using async/await: 
+    (https://node-postgres.com)
+  
+  try {
+    const res = await db.query(...,[]);
+    res.send(result.rows);
+  } catch(err) {
+    return next(err);
+  }
+
+  using promises:
+  db.query(...,[])
+    .then(res => {
+      res.send(result.rows);
+    })
+    .catch(return next(err));
+  */
+
 // Returns test string to verify that Events server is running. 
 router.get('/ping', function(req, res, next) {
-  res.send('pong - Events API');
+    res.send('pong - Events API');
 });
 
-// GET events listing. 
+// GET all events
 router.get('/', function(req, res, next) {
-    // TODO
+  db.query('SELECT * FROM event', [], (err, result) => {
+    if (err) {
+      return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+// GET all event names
+router.get('/names', function(req, res, next) {
+  db.query('SELECT fb_id, name FROM event', [], (err, result) => {
+    if (err) {
+      return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+// GET all unique event locations
+router.get('/locations', function(req, res, next) {
+  db.query('SELECT DISTINCT location_id FROM event', [], (err, result) => {
+  	if (err) {
+      return next(err);
+  	}
+  	res.send(result.rows);
+  });
+});
+
+// GET event by event_id
+router.get('/:event_id/id', function(req, res, next) {
+  const event_id = req.params.event_id;
+  db.query('SELECT * FROM event WHERE fb_id = $1', [event_id], (err, result) => {
+	if (err) {
+	  return next(err);
+	}
+	res.send(result.rows);
+  });
+});
+
+// GET all users that have favorited the given event
+router.get('/:event_id/favorites', function(req, res, next) {
+  const event_id = req.params.event_id;
+  db.query('SELECT user_id FROM favorite_events WHERE event_id = $1', [event_id], (err, result) => {
+	if (err) {
+	  return next(err);
+	}
+	res.send(result.rows);
+  });
 });
 
-module.exports = router;
+// GET all users registered and have paid or not paid for a given event
+router.get('/:event_id/register', function(req, res, next) {
+  const event_id = req.params.event_id;
+  const paid = req.query.paid;
+  if (paid == undefined){
+	db.query('SELECT user_id FROM event_registration WHERE event_id = $1', [event_id], (err, result) => {
+	  if (err) {
+        return next(err);
+      }
+      res.send(result.rows);
+  	});
+  }
+  else{
```

Comment:
if statements should have this format:

if (condition) {
    code here
} else {
   code here
}

rather than

if (condition){
    code here
} 
else{
   code here
}